core/src/main/scala/kafka/zookeeper/ZooKeeperClient.scala

```diff
@@ -59,6 +59,7 @@ class ZooKeeperClient(connectString: String,
   private val zNodeChildChangeHandlers = new ConcurrentHashMap[String, ZNodeChildChangeHandler]().asScala
   private val inFlightRequests = new Semaphore(maxInFlightRequests)
   private val stateChangeHandlers = new ConcurrentHashMap[String, StateChangeHandler]().asScala
+  private val expiryHandlerExecutor = new ThreadPoolExecutor(0, 1, 60L, TimeUnit.SECONDS, new SynchronousQueue)
```

Comment:
Could we just use KafkaScheduler? One of the benefits is that it logs unexpected exceptions while running a scheduled task. We also want to close the scheduler in close().